Penteract/Assets/Scripts/HUDController.cpp

```diff
@@ -359,23 +379,29 @@ void HUDController::UpdateCommonSkill() {
 	}
 }
 
-void HUDController::UpdateFangCooldowns(GameObject* fangSkillCanvas) {
-	std::vector<GameObject*> children = fangSkillCanvas->GetChildren();
+
+void HUDController::UpdateFangCooldowns(GameObject* fangSkillCanvas, bool isMain) {
+	std::vector<GameObject*> skills = fangSkillCanvas->GetChildren();
 	int skill = FANG_SKILL_1;
-	for (std::vector<GameObject*>::iterator it = children.begin(); it != children.end(); ++it) {
-		std::vector<GameObject*> skills = (*it)->GetChildren();
-		for (std::vector<GameObject*>::iterator itSkills = skills.begin(); itSkills != skills.end(); ++itSkills) {
-			ComponentImage* image = (*itSkills)->GetComponent<ComponentImage>();
+	for (std::vector<GameObject*>::iterator it = skills.begin(); it != skills.end(); ++it) {
+
+		if ((*it)->GetChildren().size() > 0) {
+			ComponentImage* image = (*it)->GetChildren()[1]->GetComponent<ComponentImage>();
 			if (image) {
 				if (image->IsFill()) {
 					switch (skill) {
 					case Cooldowns::FANG_SKILL_1:
+						if (isMain) {
```

Comment:
Either remove these brackets or add them in the others switch cases to keep consistency

Reply:
True